api/pkg/transformer/types/series/series.go

```diff
@@ -0,0 +1,150 @@
+package series
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/go-gota/gota/series"
+
+	"github.com/gojek/merlin/pkg/transformer/types/converter"
+)
+
+type Type string
+
+const (
+	String Type = "string"
+	Int    Type = "int"
+	Float  Type = "float"
+	Bool   Type = "bool"
+)
+
+type Series struct {
+	series *series.Series
+}
+
+func NewSeries(s *series.Series) *Series {
+	return &Series{s}
+}
+
+func New(values interface{}, t Type, name string) *Series {
+	s := series.New(values, series.Type(t), name)
+	return &Series{&s}
+}
+
+func NewInferType(values interface{}, seriesName string) (*Series, error) {
+	seriesType := detectType(values)
+	seriesValues, err := castValues(values, seriesType)
+	if err != nil {
+		return nil, err
+	}
+
+	return New(seriesValues, seriesType, seriesName), nil
+}
+
+func (s *Series) Series() *series.Series {
+	return s.series
+}
+
+func (s *Series) Type() Type {
+	return Type(s.series.Type())
+}
+
+func detectType(values interface{}) Type {
+	var hasBool, hasString, hasInt, hasFloat bool
+	v := reflect.ValueOf(values)
+	switch v.Kind() {
+	case reflect.Slice:
+		for i := 0; i < v.Len(); i++ {
+			if v.Index(i).Interface() == nil {
+				continue
+			}
+
+			switch v.Index(i).Interface().(type) {
+			case float64, float32:
+				hasFloat = true
+			case int, int8, int16, int32, int64:
+				hasInt = true
+			case bool:
+				hasBool = true
+			default:
+				hasString = true
+			}
+		}
+	default:
+		switch values.(type) {
```

Comment:
this logic seems redundant with the above one? can it is extracted in different function?

Reply:
Updated in https://github.com/gojek/merlin/pull/115/commits/07d7ad9d9297bfb40ef1dfb2e9b063c15baf4884